assets/test/spec/services/applicationGeneratorSpec.js

```diff
@@ -431,4 +450,17 @@ describe("ApplicationGenerator", function(){
       );
     });
   });
+
+  describe("generating no service or routes when the image has no exposed ports", function(){
+    var resources;
+    beforeEach(function(){
+      input.image.dockerImageMetadata.ContainerConfig.ExposedPorts = {};
```

Comment:
Yeah, it would blow up in createFromImage.js reading `ContainerConfig.Env`, but I'll fix that as well.